torchmultimodal/models/omnivore.py

```diff
@@ -137,70 +182,70 @@ def _omnivore_swin_b_encoder() -> SwinTransformer3d:
         patch_embed=PatchEmbedOmnivore,
         num_classes=None,
     )
+    if pretrained:
+        common_utils.load_module_from_url(
+            encoder,
+            _OMNIVORE_PRETRAINED_URLS["swin_b_encoder"],
+            progress=progress,
+        )
     return encoder
 
 
-def omnivore_swin_t(
-    encoder_only: bool = False, pretrained: bool = False, progress: bool = True
-) -> nn.Module:
+def omnivore_swin_t(pretrained: bool = False, progress: bool = True) -> nn.Module:
     """
     Builder function to get omnivore model with swin_t variant encoder
     Args:
-        encoder_only (bool): If true then the builder will only return encoder without head (default: False)
         pretrained (bool): If true then the it will load pretrained weight,
-            otherwise it will have random weight (default: False)
-        progress (bool): If true then there will be a progress bar for downloading weight (default: True)
+            otherwise it will have random weight (default: ``False``)
+        progress (bool): If true then there will be a progress bar for downloading weight (default: ``True``)
     """
-    encoder = _omnivore_swin_t_encoder()
+    encoder = omnivore_swin_t_encoder(pretrained=pretrained)
     heads = _multimodal_head(input_dim=encoder.num_features)
-    model = OmnivoreArchitecture(encoder, heads)
     if pretrained:
-        model.load_model(_OMNIVORE_PRETRAINED_URLS["swin_t"])
-    if encoder_only:
-        return model.encoder
-    else:
-        return model
+        common_utils.load_module_from_url(
```

Comment:
Ok. I see why you are pivoting towards the util because now you are loading the encoder and heads separately.

Reply:
yes exactly